openbsd-compat/types.h

```diff
@@ -27,8 +27,8 @@ typedef unsigned char   u_char;
 typedef unsigned short  u_short;
 typedef unsigned int    u_int;
 typedef uint32_t        in_addr_t;
-typedef uint32_t        mode_t;
-typedef uint32_t        uid_t;
+typedef unsigned short  mode_t;
+typedef long            uid_t;
```

Comment:
Given we don't use these definitions from openbsd-compat/types.h, would you agree it is better to just trim them?

Reply:
```mode_t``` is actually used by the ```fuzz/``` directory. The others are not. Ideally these shouldn't make its way directly or indirectly into the public headers.

Reply:
`mode_t` is used in `fuzz/`, but it isn't coming from openbsd-compat/types.h, as far as I can see.

Reply:
Its is included indirectly:

```
fuzz_assert.c
24:#include "../openbsd-compat/openbsd-compat.h"

fuzz_bio.c
20:#include "../openbsd-compat/openbsd-compat.h"

fuzz_cred.c
19:#include "../openbsd-compat/openbsd-compat.h"

fuzz_credman.c
20:#include "../openbsd-compat/openbsd-compat.h"

fuzz_mgmt.c
18:#include "../openbsd-compat/openbsd-compat.h"
```

```openbsd-compat.h``` includes ```types.h```:
```
#if defined(_MSC_VER)
#include "types.h"
#endif
```

Reply:
Indeed, but the `fuzz/` files including openbsd-compat.h (and therefore types.h) don't use `mode_t`. The only files in `fuzz/` using `mode_t` (preload-snoop.c and preload-fuzz.c) don't include anything but system headers, which means there is nothing using the definition from types.h, as far as I can see. Does that make sense?

Reply:
Indeed. I will remove these then, its a better fix.

Reply:
All fixed.